### PR TITLE
CLOUDP-73696: Implement backup configuration ops manager go client

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-73696

Closes #[issue number]

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Comment by andreaangiolillo** [comment]
@gssbzn tagging you here so that you can review this PR once you're back

File: opsmngr/backup_configs.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+var _ BackupConfigsService = &BackupConfigsServiceOp{}
+
+// BackupConfigs represents a Backup configuration in the MongoDB Ops Manager API
+type BackupConfig struct {
+	GroupID            string   `json:"groupId,omitempty"`
+	ClusterID          string   `json:"clusterId,omitempty"`
+	StatusName         string   `json:"statusName,omitempty"`
+	StorageEngineName  string   `json:"storageEngineName,omitempty"`
+	EncryptionEnabled  bool     `json:"encryptionEnabled,omitempty"`
+	SSLEnabled         bool     `json:"sslEnabled,omitempty"`
```
Comment (gssbzn) [inline opsmngr/backup_configs.go:51]: `omitempty` and bool should use a pointer on the bool or `false` won't be serialised

File: opsmngr/backup_configs.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	ExcludedNamespaces []string `json:"excludedNamespaces,omitempty"`
+	IncludedNamespaces []string `json:"includedNamespaces,omitempty"`
+	AuthMechanismName  string   `json:"authMechanismName,omitempty"`
+	Password           string   `json:"password,omitempty"`
+	Provisioned        bool     `json:"provisioned,omitempty"`
+	SyncSource         string   `json:"syncSource,omitempty"`
+	Username           string   `json:"username,omitempty"`
+}
+
+// BackupConfigs represents an array of BackupConfig
```
Comment (gssbzn) [inline opsmngr/backup_configs.go:61]:
```suggestion
// BackupConfigs represents a paginated collection of BackupConfig
```

File: opsmngr/backup_configs_test.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/go-test/deep"
+)
+
+func TestAutomation_GetBackupConfig(t *testing.T) {
```
Comment (gssbzn) [inline opsmngr/backup_configs_test.go:25]: [copy/paste] should this be 
```suggestion
func TestBackupConfigsServiceOp_Get(t *testing.T) {
```

File: opsmngr/backup_configs_test.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+		StorageEngineName: "WIRED_TIGER",
+		EncryptionEnabled: false,
+		SSLEnabled:        false,
+	}
+	if diff := deep.Equal(config, expected); diff != nil {
+		t.Error(diff)
+	}
+}
+
+func TestAutomation_ListBackupConfig(t *testing.T) {
```
Comment (gssbzn) [inline opsmngr/backup_configs_test.go:59]: same as above, the test method name should match the Struct, underscore, function pattern

**Review by gssbzn — CHANGES_REQUESTED** [review]
Just a couple of small things, good work

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**